### PR TITLE
Login/register functionalities added.

### PR DESCRIPTION
**Pull request by MartynasAnn**
-Login page added, restricting access to the rest of the app while not logged in.
-Register page added for creation of new players, can only be accessed via the login page.
-Registered players added to database.

### CODE REVIEW

File: EncounterMeApp/EncounterMeApp/Models/Player.cs [hunk 1]
```diff
@@ -17,6 +17,9 @@ public class Player : IComparable
         public int LocationsOwned { get; set; }
         public string ProfilePic { get; set; }
         public string Email { get; set; }
+        public string Firstname { get; set; }
+        public string Lastname { get; set; }
+        public string Password { get; set; }
```
Comment (Dominic-dul) [inline EncounterMeApp/EncounterMeApp/Models/Player.cs:22]: Maybe password should be private?

File: EncounterMeApp/EncounterMeApp/Views/LoginPage.xaml.cs [hunk 2]
```diff
@@ -6,15 +6,75 @@
...
+        {
+            PlayerList = new List<Player>();
+
+            PlayerList.Clear();
+
+            var players = await PlayerDatabase.GetPlayers();
+
+            PlayerList.AddRange(players);
+
+            /*if (Application.Current.Properties.ContainsKey("Username") && Application.Current.Properties.ContainsKey("Password"))
```
Comment (Dominic-dul) [inline EncounterMeApp/EncounterMeApp/Views/LoginPage.xaml.cs:41]: Delete the comments if not neccessary

**Review by Dominic-dul — APPROVED** [review]
Everything seems fine. Maybe in the future you should add some kind of password requirements for stronger passwords. Also You should check, if there are already players with wanted nickname, when wanting to register. All in all, validation should be added in the future

File: EncounterMeApp/EncounterMeApp/Models/Player.cs (hunk 1 above)
Comment (MartynasAnn) [reply EncounterMeApp/EncounterMeApp/Models/Player.cs:22]: Thanks, will change passwords not to be saved in plaintext in the future, as this is definetely not safe.

**Review by PijusPale — APPROVED** [review]
Took you long enough! Maybe you should delete the commented lines but other than that everything seems fine to me :)

**Pull request merged**